### PR TITLE
scertec: use acme/autocert's supportsECDSA code rather than fork it

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#13050
Updates golang/go#65727

### CODE REVIEW

**Comment by bradfitz** [comment]
I have mixed feelings about this, but I think on the whole I prefer this over copy/pasting that code? Dunno.

**Review by awly — COMMENTED** [review]
This _feels_ hacky but I'm not opposed to it.
Is there any noticeable performance overhead to this indirect check?

**Comment by creachadair** [comment]
This seems fine to me, though I would also have been fine with maintaining the fork, which seems like it was simpler. I don't have a strong opinion, but it seems like any change they make upstream we're going to have to do some updates anyway. So: LGTM both before and after.

**Review by creachadair — APPROVED** [review]

File: scertec.go [hunk 1]
```diff
@@ -132,51 +138,60 @@ func (c *Client) parsedCert(secName string, pems []byte) (*tls.Certificate, erro
...
-		if !ecdsaOK {
-			return false
-		}
+// Rather than copying acme/autocert's private implementation of this, we use
+// acme/autocert's own implementation indirectly by giving it a fake
+// autocert.Cache implementation and seeing which cache key autocert tries to
+// grab. It assumes that autocert fetches cache keys ending in "+rsa" for RSA
+// keys which in practice won't change (thanks, Hyrum!), but we also lock it
+// down in tests so we'll catch it if that behavior changes. Meanwhile,
+// discussions are underway of exporting that logic from acme/autocert
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply scertec.go:176]: There is not. Private chats.

**Comment by bradfitz** [comment]
> Is there any noticeable performance overhead to this indirect check?

Probably, but a blip in the grand scheme of everything else going on in the TLS handshake.

**Review by awly — APPROVED** [review]
LGTM, let's just watch trunkd/control response time metrics whenever this gets added to them :crossed_fingers:

File: scertec.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bradfitz) [reply scertec.go:176]: Filed https://github.com/golang/go/issues/65727. Will update the comment here.

**Pull request merged**